QemuPkg/Library/DfciDeviceIdSupportLib/DfciDeviceIdSupportLib.inf

```diff
@@ -0,0 +1,44 @@
+## @file
+# DfciDeviceIdSupportLib.inf
+#
+# NULL Library for DfciDeviceIdSupportLib.
+#
+# Copyright (C) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+#
+##
+
+[Defines]
+  INF_VERSION                    = 0x00010005
+  BASE_NAME                      = DfciDeviceIdSupportLib
+  FILE_GUID                      = 191d0ad4-469a-4d62-b526-29abe84d6af2
+  MODULE_TYPE                    = DXE_DRIVER
+  VERSION_STRING                 = 1.0
+  LIBRARY_CLASS                  = DfciDeviceIdSupportLib|DXE_DRIVER UEFI_APPLICATION
+
+
+#
+# The following information is for reference only and not required by the build tools.
+#
+#  VALID_ARCHITECTURES           = IA32 X64
+#
+
+[Sources]
+  DfciDeviceIdSupportLib.c
+
+[Packages]
+  MdePkg/MdePkg.dec
+  MdeModulePkg/MdeModulePkg.dec
+  DfciPkg/DfciPkg.dec
+
+[LibraryClasses]
+  DebugLib
+  BaseLib
+  MemoryAllocationLib
+  UefiBootServicesTableLib
+
+[Protocols]
+  gEfiSmbiosProtocolGuid
+
+[Depex]
+  TRUE
```

Comment:
nit: newline at end of file

Should `gEfiSmbiosProtocolGuid` be in the Depex? We fail out of the function if we don't find it (and aren't initialized).

Reply:
No.  Dfci is event driven.  The functions that call for the Device Id occurr  at Ready To Boot.  Putting a depex on that protocol will disturb other things.  SMBIOS should be published before Ready to Boot.